Comment View Controller Refactoring and Documentation. Networking Issue Fix.

Josiah-Agosto opened the pull request:
This Pull Request, contains CommentVC being mildly refactored and most importantly documentation added to make it readable to other developers. Along with a few other minor fixes. This PR also fixes the issue of Comments not being reloaded after a network issue. Lastly, native network framework added for better usability of monitoring network changes.

Josiah-Agosto committed:
Refactoring comment view controller, table view protocols separated. Part 1.

Josiah-Agosto committed:
Refactored comment view controller search bar delegate. Part 2.

Josiah-Agosto committed:
Refactored comment view controller reply delegate. Part 3.

Josiah-Agosto committed:
Refactored TTTAttributed Cell Delegate from Comment View Controller. Part 4.

Josiah-Agosto committed:
Minor refactoring fixes to Comment Controller and Protocols.

Josiah-Agosto committed:
Adding signing and capabilities.

Josiah-Agosto committed:
Adding signing.

Josiah-Agosto committed:
Comment view controller refactoring and added documentation. Also, minor fixes.

Josiah-Agosto committed:
Fixed a few bugs when running Slide on device.

Josiah-Agosto committed:
Merge branch 'feat/ios14' into develop

Josiah-Agosto committed:
Updated Slide for Reddit App Delegate.

Josiah-Agosto committed:
Merge branch 'feat/ios14' into develop

Josiah-Agosto committed:
Fixed pod errors after pulling.

Josiah-Agosto committed:
Merge branch 'feat/ios14' into develop

Josiah-Agosto committed:
Changing install-filter script to original.

Josiah-Agosto committed:
Removing personal info from entitlements.

Josiah-Agosto committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	Slide for Reddit.xcodeproj/xcshareddata/xcschemes/Slide for Apple Watch.xcscheme

Josiah-Agosto committed:
Refactor CommentVC file structure.

Josiah-Agosto committed:
Fixed PR issues, files were moved around and renamed.

Josiah-Agosto committed:
Fixing Run SwiftLint enabled checkbox.

Josiah-Agosto committed:
Fixing project setting.

Josiah-Agosto committed:
Removing private information.

Josiah-Agosto committed:
Extra Documentation Added to Comment View Controller.

Josiah-Agosto committed:
Initial Setup for Network Monitor.

Josiah-Agosto committed:
Adding extra missing documentation to CommentVC.

Josiah-Agosto committed:
Extra tweaking for Network Monitor.

Josiah-Agosto committed:
Fixes reloading comments after a Network Issue. Also, new native framework being used for getting Network status throughout the app with better ease.

Josiah-Agosto committed:
Fixing merge conflicts.

colejd commented on scripts/install-filter.sh:
```diff
@@ -11,8 +11,8 @@
-your_developer_id="1234567890" # Your 10 digit developer ID. Example is FTT89576VQ
-your_usr_id="anything.you.want" # Anything you want. Sets USR_DOMAIN in the project.
+your_developer_id="55T4UN5QF4" # Your 10 digit developer ID. Example is FTT89576VQ
+your_usr_id="com.JosiahAgosto" # Anything you want. Sets USR_DOMAIN in the project.
```
Please don't commit these changes. Since the script is a one-time thing, you can just remove your changes and the automatic substitution will still work.

Josiah-Agosto replied:
Omg, again I committed this. Sorry I'll get this removed again.

colejd commented on scripts/install-filter.sh:
```diff
@@ -28,4 +28,4 @@ sed -i.bak -e "s/$project_developer_id/$your_developer_id/" -e "s/$project_usr_i
 # Converts credentials from yours to project's when making commits
-git config filter.filter-credentials.clean "sed -e 's/$your_developer_id/$project_developer_id/' -e 's/$your_usr_id/$project_usr_id/'"
\ No newline at end of file
+git config filter.filter-credentials.clean "sed -e 's/$your_developer_id/$project_developer_id/' -e 's/$your_usr_id/$project_usr_id/'"
```
There's a diff but I'm having trouble seeing what's actually different here.

Josiah-Agosto replied:
Hmm, I don't see a difference at all. I'll check it through the project.

Josiah-Agosto replied:
Changes like this I haven't touched, these must've changed when I pulled. This file also matches with the origin/develop branch. I'll still look into it though.

Josiah-Agosto replied:
I put the two lines in some sort of text comparing thing and they are identical I just don't know why they are marking them as different.

colejd reviewed (changes requested):
Carlos will do actual review of your logic, but I just found one thing that you should revert.

Josiah-Agosto committed:
Update install-filter.sh

Josiah-Agosto committed:
Fixing pulled commits.

Fixing merge conflicts.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Upated Apple Watch Scheme.

Josiah-Agosto committed:
Fixing Pod errors and minor comment VC tweaks.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Implementing Notifications for Comment VC.

This commit implements Notifications as a way to update the View Controllers of Online status. There is a Retain Cycle that still needs to be fixed.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Fixing commentVC loading issue.

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -1568,7 +1636,7 @@
 					BE99C9F32378AC91007911E2 = {
 						CreatedOnToolsVersion = 11.2.1;
-						DevelopmentTeam = 68GVUVA6AU;
+						DevelopmentTeam = FTT89576VQ;
```
Just flagging some of the signing stuff that needs to be reverted

Needs to be changed back to 68GVUVA6AU

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2278,14 +2370,13 @@
 				SWIFT_VERSION = 4.2;
 				TARGETED_DEVICE_FAMILY = 4;
-				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).redditslide-debug.watchkitapp";
+				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).redditslide.watchkitapp";
```
Needs to be reverted back PRODUCT_BUNDLE_IDENTIFIER and USR_WATCHKIT_IDENTIFIER

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2310,13 +2401,13 @@
-				PRODUCT_BUNDLE_IDENTIFIER = ccrama.me.redditslide.bundledwatchkitapp;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "match Development ccrama.me.redditslide.bundledwatchkitapp";
+				PROVISIONING_PROFILE_SPECIFIER = "Reddit For Slide WatchKit 2020";
```
Needs to be reverted back  PROVISIONING_PROFILE_SPECIFIER and PRODUCT_BUNDLE_IDENTIFIER

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2345,7 +2436,7 @@
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = 4;
-				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).redditslide-debug.watchkitapp";
+				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).redditslide.watchkitapp";
```
Should be reverted, debug has a different identifier than prod

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2394,16 +2483,16 @@
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				INFOPLIST_FILE = "Open in Slide/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
This should now be 11

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2394,16 +2483,16 @@
 				SKIP_INSTALL = YES;
 				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
+				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).Slide.debug.watchkitapp.watchkitextension";
```
Git might be messing up which line this belongs to, but it should be the Open-in-Slide extension

Josiah-Agosto replied:
So it should look like this, "$(USR_DOMAIN).Open-in-Slide.watchkitextension"?

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2415,20 +2504,18 @@
 				CLANG_ENABLE_OBJC_WEAK = YES;
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Should be reverted to "iPhone Developer"

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2415,20 +2504,18 @@
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				INFOPLIST_FILE = "Open in Slide/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Deployment target should be 11 now

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2492,10 +2580,9 @@
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
 				TARGETED_DEVICE_FAMILY = "1,2";
-				USR_BASE_IDENTIFIER = "$(USR_DOMAIN).redditslide-debug";
+				USR_BASE_IDENTIFIER = "$(USR_DOMAIN).redditslide.debug";
```
Debug needs its own identifier, should be changed back to -debug

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2572,7 +2659,7 @@
 				GCC_WARN_UNUSED_PARAMETER = YES;
 				INFOPLIST_FILE = "Slide for Reddit/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 9.3;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Target should be 11

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2583,35 +2670,34 @@
 				SWIFT_SWIFT3_OBJC_INFERENCE = Default;
 				SWIFT_VERSION = 5.0;
-				VALID_ARCHS = "arm64 arm64e";
+				USR_WATCHKIT_IDENTIFIER = "$(USR_DOMAIN).Slide.debug.watchkitapp.watchkitextension";
```
VALID_ARCHS was an xcode beta bug, but should be reverted back for the final release

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2583,35 +2670,34 @@
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = "Slide for Reddit/Slide for Reddit.entitlements";
-				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "Apple Development";
```
Should be changed back to "iPhone Developer"

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2583,35 +2670,34 @@
 				GCC_WARN_UNUSED_PARAMETER = YES;
 				INFOPLIST_FILE = "Slide for Reddit/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 9.3;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Deployment target should be 11

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2622,19 +2708,17 @@
-				PRODUCT_BUNDLE_IDENTIFIER = "$(USR_BASE_IDENTIFIER)";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "match Development ccrama.me.redditslide";
+				PROVISIONING_PROFILE_SPECIFIER = "Reddit For Slide 2020";
```
Provisioning profile should be reverted

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2646,16 +2730,17 @@
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				BUNDLE_LOADER = "$(TEST_HOST)";
-				DEVELOPMENT_TEAM = 68GVUVA6AU;
+				DEVELOPMENT_TEAM = FTT89576VQ;
```
Development team needs to be reverted

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2679,15 +2764,16 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
-				DEVELOPMENT_TEAM = 68GVUVA6AU;
+				DEVELOPMENT_TEAM = FTT89576VQ;
```
Development team needs to be reverted

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2706,19 +2792,20 @@
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = 68GVUVA6AU;
+				DEVELOPMENT_TEAM = FTT89576VQ;
```
Development team needs to be reverted

ccrama commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2730,13 +2817,13 @@
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = 68GVUVA6AU;
+				DEVELOPMENT_TEAM = FTT89576VQ;
```
Development team needs to be reverted

ccrama commented on Slide for Reddit.xcodeproj/xcshareddata/xcschemes/Share to Slide.xcscheme:
```diff
@@ -42,15 +42,6 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
-      <MacroExpansion>
```
I believe this is needed for Share to Slide to function, needs testing

Josiah-Agosto replied:
I have no clue what this is and how it got removed, I will look further into it.

Josiah-Agosto replied:
So I was trying to figure out what and how this was removed, on September 11th, with the commit named "Fixed freezing when exiting some settings menus". I pulled from the remote and from that file macroExpansion was removed, it must've been added back and I didn't pull so I will try to see if updating everything can fix it.

ccrama commented on Slide for Reddit/Comments/CommentViewController.swift:
```diff
@@ -250,13 +663,14 @@ class CommentViewController: MediaViewController, UITableViewDelegate, UITableVi
         }
         do {
-            try session?.getArticles(name, sort: .new, limit: SettingValues.commentLimit, completion: { (result) in
+            try session?.getArticles(name, sort: .new, completion: { (result) in
```
Need to make sure this matches the current behavior

ccrama commented on Slide for Reddit/Comments/CommentViewController.swift:
```diff
@@ -306,41 +719,10 @@ class CommentViewController: MediaViewController, UITableViewDelegate, UITableVi
                                 self.tableView.performBatchUpdates({
                                     self.tableView.insertRows(at: paths, with: .fade)
                                 }, completion: { (_) in
-                                    self.lastY = self.tableView.contentOffset.y
```
This code is required for the "live mode new comments" bubble to show up. Was this moved? Can't find it in the PR but I might have missed it :)

Josiah-Agosto replied:
Hmm, I also didn't see most of this functions changes inside mine either even when I pulled. I just copied the contents of it to mine.

ccrama commented on Slide for Reddit/Comments/CommentViewController.swift:
```diff
@@ -2450,16 +2442,31 @@ class CommentViewController: MediaViewController, UITableViewDelegate, UITableVi
+     */
     func deleteComment(cell: CommentDepthCell) {
-        let alert = UIAlertController.init(title: "Really delete this comment?", message: "", preferredStyle: .alert)
+        let alert = UIAlertController.init(title: "Really, delete this comment?", message: "", preferredStyle: .alert)
```
Might be better to title this "Remove your comment permanently?"

ccrama commented on Slide for Reddit/Comments/CommentViewController.swift:
```diff
@@ -2812,7 +2681,7 @@ class CommentViewController: MediaViewController, UITableViewDelegate, UITableVi
             if !skipTop {
                 self.tableView.scrollToRow(at: indexPath,
-                                           at: UITableView.ScrollPosition.top, animated: false)
+                                           at: UITableView.ScrollPosition.none, animated: false)
```
This may cause the TableView to not scroll at all if the view is partially visible, should skip to the top of the screen

ccrama commented on Slide for Reddit/Comments/CommentViewController.swift:
```diff
@@ -1699,26 +1598,37 @@ class CommentViewController: MediaViewController, UITableViewDelegate, UITableVi
+    /**
+     Dismissing top view controller.
+     */
+    @objc func dismissTopController(_ sender: AnyObject) {
```
Good point, we should do some more testing here, as the comment VC may be the first child of a NavVC

ccrama commented on Slide for Reddit/PagingCommentViewController.swift:
```diff
@@ -36,12 +34,12 @@ class PagingCommentViewController: ColorMuxPagingViewController, UIPageViewContr
         if currentIndex + 1 < submissionDataSource.content.count {
             currentIndex += 1
-            let vc = CommentViewController(submission: submissionDataSource.content[currentIndex + startIndex], single: false)
+            let vc = CommentViewController(submission: submissionDataSource.content[currentIndex + startIndex])
```
Single is important to this class, need to check to make sure we don't load every comment when the PagingCommentVC appears. We only want to load comments when the user swipes pages (viewDidAppear)

ccrama reviewed (commented):
Great work here! We can discuss on how to merge in the xcodeproj files, and I am assuming the logic is not changed for any of the broken-out Comments classes. Other than that, just a few minor comments I made on a few lines.

Thank you for the PR! Glad to have you on the project.

UnknownEncrypter commented:
Thank you. I didn't notice so many different things that shouldn't be committed. I will get those fixed as quick as I can.

Josiah-Agosto committed:
Updated Files.

Josiah-Agosto committed:
Merge branch 'develop' of https://github.com/ccrama/Slide-iOS into develop

Josiah-Agosto committed:
Merge branch 'develop' of https://github.com/UnknownEncrypter/Slide-iOS into develop

Josiah-Agosto committed:
Updating Signing and other Bugs.

Josiah-Agosto committed:
Fixing Signing.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Update Package.resolved

Josiah-Agosto committed:
Further updating.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

colejd commented on Slide for Reddit.xcodeproj/project.pbxproj:
```diff
@@ -2615,8 +2714,7 @@
 				TARGETED_DEVICE_FAMILY = "1,2";
 				USR_BASE_IDENTIFIER = "$(USR_DOMAIN).redditslide-debug";
 				USR_DISPLAY_NAME = "Slide Debug";
-				USR_DOMAIN = ccrama.me;
```
Need to undo this change or the whole build process will fall apart.

Josiah-Agosto replied:
I don't know how this changed. I will add it back somehow.

Josiah-Agosto replied:
Added it back.

Josiah-Agosto committed:
More signing issues fixed.

Josiah-Agosto committed:
Merge from Upstream/Develop.

Josiah-Agosto committed:
Merge from upstream

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Merge from upstream/develop

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Reverted Latest commit.

Josiah-Agosto committed:
Revert Commit.

Josiah-Agosto committed:
Revert Commit.

Josiah-Agosto committed:
Pull from upstream.

Josiah-Agosto committed:
Merge remote-tracking branch 'upstream/develop' into develop

Josiah-Agosto committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	Slide for Reddit/CommentViewController.swift
	Slide for Reddit/Comments/CommentViewController.swift
	Slide for Reddit/SettingsIcon.swift
	Slide for Reddit/SingleSubredditViewController.swift
	Slide for Reddit/TextDisplayStackView.swift

Josiah-Agosto committed:
Pulled from upstream/develop.

Josiah-Agosto committed:
Update Packages.
